vodesfunc/rescale_ext/ignoremask.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from vskernels import Kernel
+from vsscale import ScalingArgs
+from vskernels import BorderHandling
+from vsexprtools import norm_expr
+from vstools import vs
+
+from .base import descale_rescale
+
+__all__ = ["border_clipping_mask"]
+
+
+def border_clipping_mask(
+    clip: vs.VideoNode,
+    scaling_args: ScalingArgs,
+    kernel: Kernel,
+    border_handling: BorderHandling,
+    dark_thr: float | None = None,
+    bright_thr: float | None = 1.0,
+) -> vs.VideoNode:
+    size_args = dict(
+        width=scaling_args.width if scaling_args.mode == "w" else clip.width,
+        height=scaling_args.height if scaling_args.mode == "h" else clip.height,
+    )
+
+    blank = descale_rescale(
+        clip.std.BlankClip(length=1, color=0.5, keep=True, **size_args),
+        kernel,
+        width=clip.width,
+        height=clip.height,
+        border_handling=int(border_handling),
+        **scaling_args.kwargs(),
+    )
+
+    return norm_expr(
+        [clip, blank],
+        "y 0.5 - dup 0 = 0 swap2 {bright_expr} {dark_expr} ? ?",
```

Comment:
Is this correct? As far as I see, just before `swap2` is executed, the stack is:
```
top    0
[1]    y-0.5==0
[2]    y-0.5
```
`swap2` swaps `top` with `[2]`, giving:
```
top    y-0.5
[1]    y-0.5==0
[2]    0
```
and then by the time the final `?` is about to be executed, the stack is:
```
top    if (y-0.5>0) bright else dark
[1]    y-0.5==0
[2]    0
```
thus giving:
```
top    if (0>0) y-0.5==0 else (if (y-0.5>0) bright else dark)
```
which always evaluates to the false branch.

I suspect you want one of these:
```suggestion
        "y 0.5 - dup 0 = swap {bright_expr} {dark_expr} ? 0 swap ?",
```
```suggestion
        "y 0.5 - dup 0 = not swap {bright_expr} {dark_expr} ? 0 ?",
```